src/solver/modelConverter/convertorVisitor.cpp

```diff
@@ -0,0 +1,283 @@
+
+    const char* what() const noexcept override
+    {
+        return msg_.c_str();
+    }
+
+private:
+    std::string msg_;
+};
+
+std::any ConvertorVisitor::visitIdentifier(ExprParser::IdentifierContext* context)
+{
+    for (const auto& param: model_.parameters)
+    {
+        if (param.id == context->IDENTIFIER()->getText())
+        {
+            return static_cast<Node*>(registry_.create<ParameterNode>(param.id));
+        }
+    }
+
+    for (const auto& var: model_.variables)
+    {
+        if (var.id == context->getText())
+        {
+            return static_cast<Node*>(registry_.create<VariableNode>(var.id));
+        }
+    }
+
+    /* logs.error() << "No parameter or variable found with this name: " << context->getText(); */
+    throw NoParameterOrVariableWithThisName(context->getText());
+}
+
+std::any ConvertorVisitor::visitMuldiv(ExprParser::MuldivContext* context)
+{
+    auto* left = toNodePtr(visit(context->expr(0)));
+    auto* right = toNodePtr(visit(context->expr(1)));
+
+    std::string op = context->op->getText();
+    return (op == "*") ? static_cast<Node*>(registry_.create<MultiplicationNode>(left, right))
+                       : static_cast<Node*>(registry_.create<DivisionNode>(left, right));
+}
+
+std::any ConvertorVisitor::visitFullexpr(ExprParser::FullexprContext* context)
+{
+    return context->expr()->accept(this);
+}
+
+std::any ConvertorVisitor::visitNegation(ExprParser::NegationContext* context)
+{
+    auto n = toNodePtr(context->expr()->accept(this));
+    return static_cast<Node*>(registry_.create<NegationNode>(n));
+}
+
+std::any ConvertorVisitor::visitExpression(ExprParser::ExpressionContext* context)
+{
+    return context->expr()->accept(this);
+}
+
+std::any ConvertorVisitor::visitComparison(ExprParser::ComparisonContext* context)
+{
+    auto* left = toNodePtr(visit(context->expr(0)));
+    auto* right = toNodePtr(visit(context->expr(1)));
+
+    std::string op = context->COMPARISON()->getText();
+    if (op == "=")
+    {
+        return static_cast<Node*>(registry_.create<EqualNode>(left, right));
+    }
+    else if (op == "<=")
+    {
+        return static_cast<Node*>(registry_.create<LessThanOrEqualNode>(left, right));
+    }
+    else
+    {
+        return static_cast<Node*>(registry_.create<GreaterThanOrEqualNode>(left, right));
+    }
+}
+
+std::any ConvertorVisitor::visitAddsub(ExprParser::AddsubContext* context)
+{
+    auto* left = toNodePtr(visit(context->expr(0)));
+    auto* right = toNodePtr(visit(context->expr(1)));
+
+    std::string op = context->op->getText();
+    return (op == "+") ? static_cast<Node*>(registry_.create<SumNode>(left, right))
+                       : static_cast<Node*>(registry_.create<SubtractionNode>(left, right));
+}
+
+// TODO implement this
+std::any ConvertorVisitor::visitPortField([[maybe_unused]] ExprParser::PortFieldContext* context)
+{
+    throw std::runtime_error(std::string("Node portfield not implemented yet"));
+    return std::any();
```

Comment:
the return is unreachable

Reply:
why you this method is not implemented?

Reply:
A lot of feature are out of scope of this PR and will be implemented in subsequent PRs